packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -343,6 +343,10 @@ module.exports = {
       navigateFallbackWhitelist: [/^(?!\/__).*/],
       // Don't precache sourcemaps (they're large) and build asset manifest:
       staticFileGlobsIgnorePatterns: [/\.map$/, /asset-manifest\.json$/],
+      // Precache files in public/, and allow them to be accessed directly
+      staticFileGlobs: [paths.appPublic + '/**/*'],
```

Comment:
can we whitelist the extension here? I usually have `.htaccess` and `index.php`

Reply:
the `.htaccess` will not, because it won't be picked up by the glob.

it _will_ try to download the index.php, the result of which will depend on the configuration of your server.

this highlights something that might surprise a user just as much as them being unable to access their `public/` files:

*anything* you put in `public/` will now be precached during the first load.

depending on what people are doing in their `public/` folder, this could have alarming consequences.

Reply:
in my own use case i've favoured `navigate` ignoring requests which appear to have file extensions.

but this is not ideal either, because we can't know what routes the user wants to handle in their app.

i think the least surprising behaviour would be for everything in `public/` to be ignored by `navigateFallback`. the configuration for navigateFallback is, however, an array of regexen that indicate a whitelist. we'd really rather have a blacklist.

the code would be bizarre, and could result in quite a long list of RegExps for the service worker to match against for every request.

like, you'd be setting the whitelist to something like:

```
navigateFallbackWhitelist:
  filesInPublic.map(path =>
    RegExp('^(?!' + path + ').*$')
  )
```

which is not ideal.

the dream would be if there were a navigateFallbackBlacklist.

@jeffposnick mentioned [here](https://github.com/GoogleChrome/sw-precache/issues/292#issuecomment-298697958) that that was planned for a future library

Reply:
(actually that wouldn't even work because it is checked with `whitelist.some`. so everything would pass one of the regexen.)

this might:

```js
navigateFallbackWhitelist: [ '(?!' + filesInPublic.join('|') + ').*$' ]
```

Reply:
how about just whitelist it for now?
```
staticFileGlobs: ['js', 'css', 'json', 'html', 'svg', 'otf', 'ttf'].map(ext => paths.appPublic + '/**/*.' + ext),
```
will it navigate correctly?

Reply:
ah, the non-whitelisted extensions would still not be able to be accessed... `navigateFallbackBlacklist` would be ideal...

Reply:
plus we'd be making a decision as to what files would be accessible and viewable in browser for the user, which may be quite surprising. (ie: how come i could visit my `data.json` but not my `data.txt`, how come `view image in new tab` doesn't work for `.gif`s)

@jeffposnick would you consider a PR on `sw-precache` adding a `navigateFallbackBlacklist`, with a default of `[]` that returned early if `url`  matches any of its items?

however, switching to such a blacklist would mean there'd be no way of doing what this PR would do in its current state, which is getting something into precache that isn't required in the js.

@viankakrisna what is your expected behaviour for that `index.php` in public? it surely isn't currently working once the service worker has installed?

Reply:
@chee mainly for injecting initial data and user agent redirects (i want to send server generated markup for bots). It would still work for initial request and bots (afaik no bots will install sw right?).

Reply:
ah that makes sense.

so if there was a fallback blacklist then it would behave the same as it does now for you, but with this pull request a user would be needlessly caching the result of your `index.php` running

Reply:
Yeah, and i thought it would cache the contents of index.php. Thanks for clearing that up!